python/cudf/cudf/core/dtypes.py

```diff
@@ -527,9 +552,28 @@ def to_arrow(self):
 
     @classmethod
     def from_pandas(cls, pd_dtype: pd.IntervalDtype) -> "IntervalDtype":
-        return cls(
-            subtype=pd_dtype.subtype
-        )  # TODO: needs `closed` when we upgrade Pandas
+        return cls(subtype=pd_dtype.subtype, closed=pd_dtype.closed)
```

Comment:
The conda environment still just asks for pandas >= 1.0, which is not enough, but the version that was installed has the `closed` property. I can drop this part if 1.0 is actually still a compatible pandas version.

Reply:
I believe it is not? cc: @galipremsagar

Reply:
Yea, `closed` was introduced in `1.3` version of pandas. Maybe we should version this.

Reply:
This needs more handling in any case since a pandas intervaldtype can have `None` as closed (meaning default), whereas the cudf one defaults to `right`.

Reply:
In the interests of reducing feature-creep, I'll pull this change into a separate PR